### PR TITLE
PWGHF: Speedup loops and preselections

### PR DESCRIPTION
**Pull request by fgrosa**
- apply to 2 prongs same improvements implemented for 3 prongs in https://github.com/AliceO2Group/O2Physics/pull/47 and https://github.com/AliceO2Group/O2Physics/pull/39
- use partitions to loop only over positive/negative tracks

It was tested and it gives exactly the same result as the previous version (but it should be faster)

### CODE REVIEW

**Review by vkucera — DISMISSED** [review]

File: Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx [hunk 1]
```diff
@@ -539,16 +539,17 @@ struct HfTrackIndexSkimsCreator {
   }
 
   /// Method to perform selections for 2-prong candidates before vertex reconstruction
-  /// \param hfTracks is the array of 2 tracks
+  /// \param hfTrack0 is the first daughter track
+  /// \param hfTrack1 is the second daughter track
   /// \param massHypos is a 2D array containing the mass hypotheses for the 2-prong channels
   /// \param cutStatus is a 2D array with outcome of each selection (filled only in debug mode)
```
Comment (vkucera) [inline Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx:545]: Can you please fix the documentation as well?
`massHypos` seems confused with `whichHypo`

**Comment by vkucera** [comment]
@fgrosa Could you please also implement the changes from https://github.com/AliceO2Group/O2Physics/pull/52 ?

File: Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx (hunk 1 above)
Comment (fgrosa) [reply Tasks/PWGHF/HFTrackIndexSkimsCreator.cxx]: Done!

**Comment by fgrosa** [comment]
> @fgrosa Could you please also implement the changes from #52 ?

Done! :thumbsup:

**Review by vkucera — APPROVED** [review]

**Pull request merged**